lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -282,7 +282,7 @@ bool hasConvertToMMATransisitiveUse(Operation *op, Attribute encoding) {
 bool isLayoutAnchor(Operation *op) {
   if (isa<LoadOp, StoreOp>(op))
     return isExpensiveLoadOrStore(op);
-  if (isa<DotOp, nvidia_gpu::DotAsyncOp, AtomicRMWOp, AtomicCASOp>(op))
+  if (isa<DotOp, nvidia_gpu::DotAsyncOp, AtomicCASOp>(op))
```

Comment:
That will undo the coalescing done in coalescing oass. How is it always the right thing to do?
The point of the anchors is that they already have a layout picked to have good memory access patterns. It sounds like in some cases this would generate suboptimal code.

Reply:
This is an interesting topic actually.

Prior to sm90, atomic instructions do not have good vector type support. So the maximum vector length in triton (nvidia backend) is just f16x2.

However, since sm90, we could use similar vector modifiers as load instructions.

> Support for vector types requires sm_90 or higher.
> .vec_16_bit =         { .v2, .v4, .v8 }
> .vec_32_bit =         { .v2, .v4 };

Reply:
even without vector support L1 coalescing still applies right?

Reply:
Yeah that's a good point. Maybe we need to take a look at the IR and then draft some heuristics to determine whether or not to convert layout.